src/components/tab-panel/examples/tab-panel-content.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+    Component,
+    h,
+    Prop,
+    EventEmitter,
+    Event,
+    State,
+    Watch,
+} from '@stencil/core';
+import { Tab } from '@limetech/lime-elements';
+
+const LOAD_TIME = 1500;
+
+@Component({
+    tag: 'limel-example-tab-panel-content',
+    shadow: true,
+    styleUrl: 'tab-panel-content.scss',
+})
+export class TabPanelContentExample {
+    @Prop()
+    public tab: Tab;
+
+    @Event()
+    private changeTab: EventEmitter<Tab>;
+
+    @State()
+    private votes = 0;
+
+    @State()
+    private loaded = false;
+
+    constructor() {
+        this.vote = this.vote.bind(this);
+    }
+
+    /**
+     * Load the tab contents the first time the tab is activated
+     */
+    @Watch('tab')
+    protected watchTab() {
+        if (!this.loaded && this.tab.active) {
+            this.loadTabContent();
+        }
+    }
+
+    /**
+     * Simulate loading the tab content
+     */
+    private loadTabContent() {
+        setTimeout(() => {
+            this.loaded = true;
+        }, LOAD_TIME);
+    }
+
+    public render() {
```

Comment:
I think functions should be sorted with public before protected, and protected before private. The consultants look very much to our documentation and examples for guidance, so I think it's important to keep the examples as clean and well-formatted as any production code.

Reply:
Fixed! However, I don't agree that there should be a general rule to order stuff in public > protected > private order. Code that belong together should stay together :slightly_smiling_face: If a public method uses a private one, I should not have to scroll to the bottom just to read it :slightly_smiling_face: There should be a natural flow of reading the code if possible

Reply:
I both agree and disagree. All rules can be broken if the circumstances justify it. But I think there's value in putting the stuff that constitutes the external interface bundled together at the top. That gives the reader of the code a quick overview of the component and what it does. Any non-trivial implementation details are typically quite nice to put in well-named private functions further down. Whatever specific detail you are interested in, you can just jump to in any modern editor.

Of course it's a trade off, and I'm not saying it's a rule set in stone. But I think it's a good guideline.